src/regdesk/states/loading_state.js

```diff
@@ -0,0 +1,51 @@
+import { RegState } from './reg_state.js';
+
+/**
+ * Loading state
+ */
+export class LoadingState extends RegState {
```

Comment:
One of these days you'll learn inheritance is evil and should be avoided at all costs.

Reply:
This is pretty much the only time I ever use it anymore: there's a handful of behaviors that everything needs to do the same and everything else is implementation specific.

Most of my C# these days is all classes with a batch of interfaces, but JS doesn't have interfaces.

Proper polymorphism-based shenanigans have come back to haunt me too many times to go looking to implement them anymore.